File: backend/app/schemas/analytics.py
"""Analytics schemas."""
from typing import Optional
from pydantic import BaseModel


class SentimentCount(BaseModel):
    positive: int
    neutral: int
    negative: int


class SentimentPercentage(BaseModel):
    positive: float
    neutral: float
    negative: float


class ScoreDistributionBucket(BaseModel):
    range: str
    count: int


class AspectAnalytics(BaseModel):
    name: str
    count: int
    avg_score: float
    sentiment: str  # overall dominant sentiment


class KeywordItem(BaseModel):
    keyword: str
    count: int
    sentiment: str


class ProductSentiment(BaseModel):
    product_name: str
    total_reviews: int
    avg_score: float
    positive_count: int
    neutral_count: int
    negative_count: int
    positive_pct: float
    neutral_pct: float
    negative_pct: float


class AnalyticsResponse(BaseModel):
    dataset_id: str
    total_reviews: int
    completed_reviews: int
    sentiment_counts: SentimentCount
    sentiment_percentages: SentimentPercentage
    average_ai_score: float
    score_distribution: list[ScoreDistributionBucket]
    top_aspects: list[AspectAnalytics]
    top_positive_keywords: list[KeywordItem]
    top_negative_keywords: list[KeywordItem]
    products: list[ProductSentiment]

File: frontend/src/types/analytics.ts
export interface SentimentCount {
  positive: number;
  neutral: number;
  negative: number;
}

export interface SentimentPercentage {
  positive: number;
  neutral: number;
  negative: number;
}

export interface ScoreDistributionBucket {
  range: string;
  count: number;
}

export interface AspectAnalytics {
  name: string;
  count: number;
  avg_score: number;
  sentiment: string;
}

export interface KeywordItem {
  keyword: string;
  count: number;
  sentiment: string;
}

export interface ProductSentiment {
  product_name: string;
  total_reviews: number;
  avg_score: number;
  positive_count: number;
  neutral_count: number;
  negative_count: number;
  positive_pct: number;
  neutral_pct: number;
  negative_pct: number;
}

export interface AnalyticsResponse {
  dataset_id: string;
  total_reviews: number;
  completed_reviews: number;
  sentiment_counts: SentimentCount;
  sentiment_percentages: SentimentPercentage;
  average_ai_score: number;
  score_distribution: ScoreDistributionBucket[];
  top_aspects: AspectAnalytics[];
  top_positive_keywords: KeywordItem[];
  top_negative_keywords: KeywordItem[];
  products: ProductSentiment[];
}
